app/controllers/sites/emails_controller.rb

```diff
@@ -1,44 +1,89 @@
+# frozen_string_literal: true
+
 class Sites::EmailsController < Sites::BaseController
+  include System::UrlHelpers.system_url_helpers
 
   provider_required
 
-  activate_menu :audience, :messages, :email
-
   before_action :find_account
-  before_action :find_services
+  before_action :find_exceptions, only: :edit
   prepend_before_action :deny_on_premises_for_master
 
-  def edit
-  end
+  helper_method :props, :popover_props, :exceptions, :products_with_default_email
 
-  def update
-    unless @account.update(params[:account])
-      not_saved = true
-    end
+  activate_menu :audience, :messages, :email
 
-    @services.each do |service|
-      unless service.update :support_email => params["service_#{service.id}_support_email"]
-        not_saved = true
-      end
-    end
+  def edit; end
 
-    if not_saved
-      flash.now[:warning] = t('.warning')
+  def update
+    account_saved = update_account
+    services_saved = update_services
+
+    if account_saved && services_saved
+      flash[:success] = t('.success')
+      redirect_to action: :edit
     else
-      flash.now[:success] = t('.success')
+      flash.now[:warning] = t('.warning')
+      render :edit
     end
-
-    render 'edit'
   end
 
   private
 
+  def presenter
+    @presenter ||= Api::ServicesIndexPresenter.new(user: current_user)
+  end
+
+  def update_account
+    account_params = params.require(:account).permit(%i[support_email finance_support_email services])
+
+    @account.update(account_params)
+  end
+
+  def update_services # rubocop:disable Metrics/AbcSize
+    service_params = params.require(:account).permit(services: :support_email)
+    services_valid = true
+
+    ids = service_params[:services].keys.map(&:to_i)
+    @exceptions = current_user.accessible_services.where(id: ids)
```

Comment:
I think `exceptions` doesn't need to be an instance variable here, as it's only used in the context of this method.